itests/ipni_publish_test.go

```diff
@@ -0,0 +1,109 @@
+package itests
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/filecoin-project/boost/itests/framework"
+	"github.com/filecoin-project/boost/testutil"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/itests/kit"
+	"github.com/google/uuid"
+	"github.com/ipfs/go-cid"
+	"github.com/ipni/ipni-cli/pkg/adpub"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIPNIPublish(t *testing.T) {
+	ctx := context.Background()
+	log := framework.Log
+
+	kit.QuietMiningLogs()
+	framework.SetLogLevel()
+	var opts []framework.FrameworkOpts
+	opts = append(opts, framework.EnableLegacyDeals(true))
+	f := framework.NewTestFramework(ctx, t, opts...)
+	err := f.Start()
+	require.NoError(t, err)
+	defer f.Stop()
+
+	// Get the listen address
+	addrs, err := f.Boost.NetAddrsListen(ctx)
+	require.NoError(t, err)
+
+	// Create new ipni-cli client
+	ipniClient, err := adpub.NewClient(addrs, adpub.WithEntriesDepthLimit(100000))
+	require.NoError(t, err)
+
+	// Get head when boost starts
+	headAtStart, err := ipniClient.GetAdvertisement(ctx, cid.Undef)
+	require.NoError(t, err)
+
+	err = f.AddClientProviderBalance(abi.NewTokenAmount(1e15))
+	require.NoError(t, err)
+
+	// Create a CAR file
+	tempdir := t.TempDir()
+	log.Debugw("using tempdir", "dir", tempdir)
+
+	fileSize := 200000
+	randomFilepath, err := testutil.CreateRandomFile(tempdir, 5, fileSize)
+	require.NoError(t, err)
+
+	// create a dense carv2 for deal making
+	rootCid, carFilepath, err := testutil.CreateDenseCARv2(tempdir, randomFilepath)
+	require.NoError(t, err)
+
+	// Start a web server to serve the car files
+	log.Debug("starting webserver")
+	server, err := testutil.HttpTestFileServer(t, tempdir)
+	require.NoError(t, err)
+	defer server.Close()
+
+	// Create a new dummy deal
+	log.Debug("creating dummy deal")
+	dealUuid := uuid.New()
+
+	// Make a deal
+	res, err := f.MakeDummyDeal(dealUuid, carFilepath, rootCid, server.URL+"/"+filepath.Base(carFilepath), false)
+	require.NoError(t, err)
+	require.True(t, res.Result.Accepted)
+	log.Debugw("got response from MarketDummyDeal", "res", spew.Sdump(res))
+
+	time.Sleep(2 * time.Second)
+
+	// Wait for the deal to be added to a sector
+	err = f.WaitForDealAddedToSector(dealUuid)
+	require.NoError(t, err)
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Use `require.Eventually`. Ditto re other `time.Sleep`s.

Reply:
I am removing the sleep as we don't really need them.